src/utils/useHydrateAtoms.ts

```diff
@@ -0,0 +1,25 @@
+import { useContext, useMemo } from 'react'
+import { SECRET_INTERNAL_getStoreContext as getStoreContext } from 'jotai'
+import type { Atom, Scope } from '../core/atom'
+
+export function useHydrateAtoms(
+  values: Iterable<readonly [Atom<unknown> & { hydrated?: Symbol }, unknown]>,
+  scope?: Scope
+) {
+  const StoreContext = getStoreContext(scope)
+  const restoreAtoms = useContext(StoreContext)[3]
+
+  useMemo(() => {
+    const tuplesToRestore = []
+    for (const tuple of values) {
+      const atom = tuple[0]
+      if (atom.hydrated !== hydratedSymbol) {
+        tuplesToRestore.push(tuple)
+        atom.hydrated = hydratedSymbol
```

Comment:
This is changing the atom config, which is shared by other stores/providers.
If we would like to check this, we should use `WeakMap`.
```ts
type HydratedAtoms = WeakSet<Atom<unknown>>
type HydratedMap = WeakMap<Store, HydratedAtoms>
```

Reply:
Check new commit.